lib/features/presentation/home/screen/home_screen.dart

```diff
@@ -0,0 +1,71 @@
+
+import 'package:ecommerce_app/core/resources/assets_manager.dart';
+import 'package:ecommerce_app/core/resources/color_manager.dart';
+import 'package:ecommerce_app/features/presentation/home/tabs/categories_tap/screen/categories_tab.dart';
+import 'package:ecommerce_app/features/presentation/home/tabs/home_tap/screen/home_tab.dart';
+import 'package:ecommerce_app/features/presentation/home/tabs/profile_tab/screen/profile_tab.dart';
+import 'package:ecommerce_app/features/presentation/home/tabs/wish_list_tab/screen/wish_list_tab.dart';
+import 'package:flutter/material.dart';
+
+class HomeScreen extends StatefulWidget {
+  HomeScreen({super.key});
+
+  @override
+  State<HomeScreen> createState() => _HomeScreenState();
+}
+
+class _HomeScreenState extends State<HomeScreen> {
+  int currentIndex = 0;
+  List<Widget> tabs = const [HomeTab() , CategoriesTab() , WishListTab() , ProfileTab()];
+  @override
+  Widget build(BuildContext context) {
+
+    return Scaffold(
+        extendBody: false,
+        body: tabs[currentIndex],
+        bottomNavigationBar: ClipRRect(
+          borderRadius: const BorderRadius.only(
+              topLeft: Radius.circular(15), topRight: Radius.circular(15)),
+          child: BottomNavigationBar(
+            currentIndex: currentIndex,
+            onTap: (value) => changeSelectedIndex(value),
+            backgroundColor: ColorManager.primary,
+            type: BottomNavigationBarType.fixed,
+            selectedItemColor: ColorManager.primary,
+            unselectedItemColor: ColorManager.white,
+            showSelectedLabels: false, // Hide selected item labels
+            showUnselectedLabels: false, // Hide unselected item labels
+            items: [
+              // Build BottomNavigationBarItem widgets for each tab
+              buildBottomNavigationBarItem(IconsAssets.icHome, "Home"),
+              buildBottomNavigationBarItem(IconsAssets.icCategory, "Category"),
+              buildBottomNavigationBarItem(IconsAssets.icWithList, "WishList"),
+              buildBottomNavigationBarItem(IconsAssets.icProfile, "Profile"),
+            ],
+          ),
+        ));
+  }
+
+  changeSelectedIndex(int selectedIndex) {
+    setState(() {
+      currentIndex = selectedIndex;
+    });
+  }
+
+  // Helper function to create a BottomNavigationBarItem
+  BottomNavigationBarItem buildBottomNavigationBarItem(
+      String iconPath, String title) {
+    return BottomNavigationBarItem(
+        label: title, // Label for accessibility (not displayed)
+        activeIcon: CircleAvatar(
+            backgroundColor: ColorManager.white, // Background of active icon
+            child: ImageIcon(
+              AssetImage(iconPath ),
+              color: ColorManager.primary,// Active icon imagecolor: ColorManager.primary, // Active icon color
```

Comment:
this will cause the widget to be rebuild every state change

Reply:
There was a problem and while I was searching for the Class BottomNavigationBarItem I found that it is not a widget, but it turned out that this is not correct. I will now create a class that extends the BottomNavigationBarItem and performs the same role.
Thanks for Feedback